presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java

```diff
@@ -964,6 +964,19 @@ public void testTableSampleSystemBoundaryValues()
         assertEquals(emptySample.getMaterializedRows().size(), 0);
     }
 
+    @Test
+    public void testTableSampleFiltering()
+    {
+        MaterializedResult emptySample = computeActual("SELECT DISTINCT orderkey, orderdate FROM orders TABLESAMPLE BERNOULLI (99) WHERE orderkey BETWEEN 0 AND 0");
+        MaterializedResult halfSample = computeActual("SELECT DISTINCT orderkey, orderdate FROM orders TABLESAMPLE BERNOULLI (50) WHERE orderkey BETWEEN 0 AND 9999999999");
```

Comment:
as discussed offline; please check whether WHERE clause is significant here

maybe add tests without filter to `testTableSampleSystemBoundaryValues` (renaming it to `testTableSampleSystem`)

and keep your tests with filter here (renaming to `testTableSampleWithFilter`)

Reply:
Looks like the WHERE is not significant, but I left them in and added another 50% ratio example to the other test.